salt/utils/zeromq.py

```diff
@@ -80,8 +80,11 @@ def check_ipc_path_max_len(uri):
 
 def ip_bracket(addr):
     '''
-    Convert IP address representation to ZMQ (URL) format. ZMQ expects
-    brackets around IPv6 literals, since they are used in URLs.
+    Ensure IP addresses are URI-compatible - specifically, add brackets
+    around IPv6 literals if they are not already present.
     '''
+    addr = str(addr)
+    addr = addr.lstrip('[')
+    addr = addr.rstrip(']')
```

Comment:
Isn't it possible to have a partially bracketed address? I don't remember the exact syntax, but can't you have something like `[::]:beef`?

Reply:
To answer my question, [nope](https://en.wikipedia.org/wiki/IPv6_address#Literal_IPv6_addresses_in_network_resource_identifiers), not according to Wikipedia 👍

Reply:
We're assuming this is just an ip address (no port attached) :)